src/site/layout/mod.rs

```diff
@@ -8,7 +8,7 @@ use crate::config::{analytics, theme, Config};
 use crate::errors::*;
 use axohtml::{html, text, unsafe_text};
 
-pub fn build(config: &Config, content: String) -> Result<String> {
+pub fn build(config: &Config, content: String, need_js: bool) -> Result<String> {
```

Comment:
we should add needs_js to the Page struct, similar to "is index" and use it from there (also we should standardize on `needs_js`, theres 2 spellings in this PR)

Reply:
my bad, changing everything to needs_js